src/components/Listings/index.js

```diff
@@ -1,26 +1,53 @@
-import React from "react"
+import React, { useState, useMemo } from "react"
 import { Link } from "gatsby"
 import { connectToSpreadsheet } from "react-google-sheet-connector"
 import classnames from "classnames"
+import { useDebounce } from "use-debounce"
 
 import { TYPES, OFFERS_SHEET_NAME } from "../../utils/listingUtils"
-import Listing from "../Listing"
+import useTextSearch from "../../utils/useTextSearch"
+import ListingResults from "../ListingResults"
 import cs from "./styles.module.css"
 
+const FULL_TEXT_SEARCH_KEYS = [
+  "name:",
+  "pleaseCheckOffWhatYouCanOffer:",
+  "areThereAnySuppliesOrFoodThatYouCouldContributeToACommunalPool?PleaseBeSpecificInQuantity",
+]
+
 const Listings =
   typeof window !== `undefined` &&
   connectToSpreadsheet(props => {
-    const [typeFilter, setTypeFilter] = React.useState(TYPES[0])
+    const [typeFilter, setTypeFilter] = useState(TYPES[0])
+    const [searchTerm, setSearchTerm] = useState("")
+    const [debouncedSearchTerm] = useDebounce(searchTerm, 250)
+
     let filters = {}
 
     if (typeFilter !== TYPES[0]) {
       filters["typeOfSupport?"] = typeFilter
     }
 
-    const listings = props
-      .getSheet(OFFERS_SHEET_NAME)
-      .getData()
-      .filter(l => !!l["name:"])
+    // FIXME: choose correct deps to update this memoized result
+    // currently only updated once on initial render
+    const listings = useMemo(() => {
+      let keyIndex = 0
+      return (
+        props
+          .getSheet(OFFERS_SHEET_NAME)
+          .getData()
+          .filter(l => !!l["name:"])
+          // assign a stable unique key for each listing
+          .map(l => ({ ...l, key: `listing-${keyIndex++}` }))
```

Comment:
you could also do `.map((l, i) => ({...l, key: `listing-${i}`}))`

Reply:
Silly of me... I fixed it.